tests/pipeline/test_multiomics_trainer.py

```diff
@@ -0,0 +1,224 @@
+        gcn_lr=gcn_lr,
+        vcdn_lr=vcdn_lr,
+    )
+
+    return trainer
+
+
+def test_init(test_model):
+    assert isinstance(test_model.dataset, SparseMultiOmicsDataset)
+    assert test_model.num_modalities == 3
+    assert test_model.num_classes == 2
+    assert all(isinstance(encoder, MogonetGCN) for encoder in test_model.modality_encoder)
+    assert all(isinstance(decoder, LinearClassifier) for decoder in test_model.modality_decoder)
+    assert isinstance(test_model.loss_fn, CrossEntropyLoss)
+    assert isinstance(test_model.multi_modality_decoder, VCDN)
+    assert test_model.train_multi_modality_decoder
+    assert test_model.gcn_lr == 5e-4
+    assert test_model.vcdn_lr == 1e-3
+
+
+def test_configure_optimizers(test_model):
+    optimizers = test_model.configure_optimizers()
+    assert len(optimizers) == 4
+    assert all(isinstance(optimizer, torch.optim.Adam) for optimizer in optimizers)
+    for modality in range(test_model.num_modalities):
+        assert optimizers[modality].param_groups[0]["lr"] == 5e-4
+    assert optimizers[test_model.num_modalities].param_groups[0]["lr"] == 1e-3
+
+
+@pytest.mark.parametrize("multi_modality", [False, True])
+def test_forward(test_model, multi_modality):
+    x = []
+    adj_t = []
+    for modality in range(test_model.num_modalities):
+        data = test_model.dataset.get(modality)
+        x.append(data.x[data.train_idx])
+        adj_t.append(data.adj_t_train)
+
+    outputs = test_model.forward(x, adj_t, multi_modality)
+
+    assert isinstance(outputs, list) != multi_modality
+    if not multi_modality:
+        assert len(outputs) == test_model.num_modalities
+        for modality in range(test_model.num_modalities):
+            assert outputs[modality].shape == (test_model.dataset.get(modality).num_train, test_model.num_classes)
+    else:
+        assert isinstance(outputs, torch.Tensor)
+        assert outputs.shape == (test_model.dataset.get(0).num_train, test_model.num_classes)
+        test_model.multi_modality_decoder = None
+        with pytest.raises(TypeError):
+            _ = test_model.forward(x, adj_t, multi_modality)
+
+
+def test_pipeline(test_model):
+    set_seed(2023)
+    trainer_pretrain = pl.Trainer(max_epochs=2, gpus=0, enable_model_summary=False,)
+
+    trainer_pretrain.fit(test_model)
+    result = trainer_pretrain.test(test_model)
+
+    assert 0 <= result[0]["Accuracy"] <= 1
+    assert 0 <= result[0]["F1"] <= 1
+    assert 0 <= result[0]["AUC"] <= 1
+
+    assert repr(test_model) is not None
+    assert len(repr(test_model)) > 0
+
+
+def test_multi_class_example(test_model_multi_class):
+    set_seed(2023)
+    trainer_pretrain = pl.Trainer(max_epochs=2, gpus=0, enable_model_summary=False,)
+
+    trainer_pretrain.fit(test_model_multi_class)
+    result = trainer_pretrain.test(test_model_multi_class)
+
+    assert 0 <= result[0]["Accuracy"] <= 1
+    assert 0 <= result[0]["F1 weighted"] <= 1
+    assert 0 <= result[0]["F1 macro"] <= 1
```

Comment:
A quite long test. Possible that some are already tested elsewhere?

Reply:
This is an integration test, and the trainer has not been fitted elsewhere to get and test results.